jax/_src/lax/lax.py

```diff
@@ -2709,11 +2709,16 @@ def broadcast_in_dim(operand: ArrayLike, shape: Shape,
   See Also:
     jax.lax.broadcast : simpler interface to add new leading dimensions.
   """
-  # TODO(dfm): Re-write this as a "reshard" when only the sharding changes.
   out_sharding = canonicalize_sharding(out_sharding, 'broadcast_in_dim')
-  if (np.ndim(operand) == len(shape) and not len(broadcast_dimensions) and
-      isinstance(operand, Array) and out_sharding is None):
-    return operand
+  ndim = np.ndim(operand)
+  if (ndim == len(shape) and not len(broadcast_dimensions) and
+      isinstance(operand, Array)):
+    if out_sharding is None:
+      return operand
+    else:
+      out_sharding = out_sharding.with_spec(
```

Comment:
Why do you need to normalize here? reshard should do that for you right?

Reply:
can we call `reshard` here instead of `reshard_p.bind` then? Or is that not possible due to some reasons?

Reply:
Okay, sounds good!

I just don't like that normalization logic floating around everywhere but it's fine here I guess.